be/src/storage/lake/async_delta_writer.cpp

```diff
@@ -83,11 +83,12 @@ AsyncDeltaWriterImpl::~AsyncDeltaWriterImpl() {
 }
 
 inline int AsyncDeltaWriterImpl::execute(void* meta, bthread::TaskIterator<AsyncDeltaWriterImpl::Task>& iter) {
+    auto async_writer = static_cast<AsyncDeltaWriterImpl*>(meta);
+    auto delta_writer = async_writer->_writer.get();
     if (iter.is_queue_stopped()) {
+        delta_writer->close();
```

Comment:
Call `delta_writer->close()` from `AsyncDeltaWriterImpl::execute()` while `AsyncDeltaWriterImpl::execute()` is executed in pthread.